airflow/utils/file.py

```diff
@@ -108,46 +194,77 @@ def open_maybe_zipped(fileloc, mode='r'):
         return open(fileloc, mode=mode)
 
 
-def find_path_from_directory(base_dir_path: str, ignore_file_name: str) -> Generator[str, None, None]:
+def _find_path_from_directory(
+    base_dir_path: str,
+    ignore_file_name: str,
+    ignore_rule_type: Type[_IgnoreRule],
+) -> Generator[str, None, None]:
     """
-    Search the file and return the path of the file that should not be ignored.
-    :param base_dir_path: the base path to be searched for.
-    :param ignore_file_name: the file name in which specifies a regular expression pattern is written.
+    Recursively search the base path and return the list of file paths that should not be ignored by
+    regular expressions in any ignore files at each directory level.
+    :param base_dir_path: the base path to be searched
+    :param ignore_file_name: the file name containing regular expressions for files that should be ignored.
+    :param ignore_rule_type: the concrete class for ignore rules, which implements the _IgnoreRule interface.
 
-    :return : file path not to be ignored.
+    :return : a generator of file paths which should not be ignored.
```

Comment:
```suggestion
    :return: a generator of file paths which should not be ignored.
```

Shouldn’t have that space.